app/src/main/java/com/oracle/javafx/scenebuilder/app/DocumentWindowController.java

```diff
@@ -2355,6 +2362,37 @@ private PreferencesRecordGlobal getPreferencesRecordGlobal() {
         }
         return recordGlobal;
     }
+
+    public void showMissingTypesNotificationIfNeeded(File fxmlFile) {
+        FXOMDocument doc = getEditorController().getFxomDocument();
+        if (doc.hasUnresolvableTypes()) {
+            notifyUserAboutUnresolvableImports(fxmlFile, doc.getUnresolvableTypes(), getStage());
+        }
+    }
+    
+    private void notifyUserAboutUnresolvableImports(File fxmlFile, List<String> missingTypes, Stage owner) {
+        LOGGER.log(Level.WARNING, "Detected unresolved types in FXML: {0}: {1}",
+                               new Object[] {fxmlFile, missingTypes.stream().collect(Collectors.joining(","))});
+
+        final ErrorDialog errorDialog = new ErrorDialog(owner);
+        String first10 = missingTypes.stream()
+                                     .limit(10)
+                                     .collect(Collectors.joining(";"+System.lineSeparator()));
+
+        errorDialog.setMessage(I18N.getString("alert.open.failure.unresolved.imports",
+                               Integer.toString(missingTypes.size())));
+        errorDialog.setDetails(I18N.getString("alert.open.failure.unresolved.imports.details", first10));
+        errorDialog.setDetailsTitle(I18N.getString("alert.open.failure.unresolved.imports",
+                                    Integer.toString(missingTypes.size())));
+
+        String allMissing = missingTypes.stream()
+                                        .collect(Collectors.joining(";"+System.lineSeparator()));
+
+        errorDialog.setDebugInfo(I18N.getString("alert.open.failure.unresolved.imports.advice",
+                                 allMissing, missingTypes.size()));
+        errorDialog.setTitle(I18N.getString("alert.title.open") + ": " + fxmlFile.getName());
+        errorDialog.showAndWait();
```

Comment:
Suggestion: add a button to the dialog with an `Open FXML/JAR Manager` option, so the user can add the missing dependency.